### ISSUE TITLE
Auto-generate Lua API docs for DFHack module functions

### ISSUE THREAD

**Issue by lethosor** [DFHack/dfhack#1553]
A lot of newer module functions are missing from `Lua API.rst`. Coming up with some way to auto-generate these would be nice. (In the future, it would be nice to have C++ declarations available too, although probably not in the Lua API file.)

**Comment by PatrikLundell** [comment]
Rather than auto generation, I'd suggest auto indication, i.e. have a tool list the ones that are missing (and where they come from), as a human brain ought to have a better chance of providing a useful description (not an excuse for poor function names, of course).
I'd also want the tool to indicate when something is listed but not actually present (ought to be rare, but could happen if something is renamed or the documentation, for some reason, is one step ahead of the code and thus describes something that didn't make it in yet).

The Santa list would include a parameter check as well, to catch when new parameters are added/the set modified.

**Comment by lethosor** [comment]
Yeah, that would work too. My thought was to have the documentation live in the C++ source files and just be collected at build time (maybe with some automation to generate the proper signatures for Lua, etc), since a similar approach seems to have worked well at keeping script docs up-to-date. I guess it wouldn't work well for modules like `dfhack.internal` in Lua that don't correspond directly to a C++ module, though.

We do also have `devel/dump-rpc` and `travis/check-rpc.py`, which check all of the registered RPC procedures at runtime and verify them against documentation in the `.proto` files, respectively. Something similar might work to check for undocumented functions, if figuring out exactly what functions are available to Lua from looking at the C++ files is hard. (Actually, you might be right that just indicating missing docs is more feasible...)
